pkg/apis/core/validation/validation.go

```diff
@@ -4877,8 +4963,14 @@ func ValidateReplicationController(controller *core.ReplicationController, opts
 // ValidateReplicationControllerUpdate tests if required fields in the replication controller are set.
 func ValidateReplicationControllerUpdate(controller, oldController *core.ReplicationController, opts PodValidationOptions) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&controller.ObjectMeta, &oldController.ObjectMeta, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateReplicationControllerSpec(&controller.Spec, field.NewPath("spec"), opts)...)
-	return allErrs
+	controllerClone := controller.DeepCopy()
+	CleanPodTemplateSpec(controllerClone.Spec.Template)
```

Comment:
here only the controller Spec needs to be deep copied

Reply:
done